src/lib.rs

```diff
@@ -146,6 +144,34 @@ macro_rules! probe(
 /// ```
 #[macro_export]
 macro_rules! probe_lazy(
+    (extern $semaphore:path, $provider:ident, $name:ident $(, $arg:expr)* $(,)?)
+    => ({
+        // Type-check the location to have 'static lifetime, so it is safe to
+        // put it in the stapsdt note.  The program cannot really do
+        // anything with it---the only available method is enabled(); therefore,
+        // even though not placing it in the .probes section could upset the tracing
+        // tool, it cannot be used to trigger undefined behavior.
```

Comment:
This is worth pointing out&mdash;here I assume that the tools are robust enough and will either complain or ignore the probe, but not mess themselves up.